js/navbar.js

```diff
@@ -1,12 +1,17 @@
-var lastId,
+var lastScrollId,
+    lastSubId,
     topMenu = $("#navOptions"),
     topMenuHeight = topMenu.outerHeight(),
     // All list items
     menuItems = topMenu.find("a"),
     // Anchors corresponding to menu items
     scrollItems = menuItems.map(function(){
       var item = $($(this).attr("href"));
-      if (item.length) { return item; }
+      if (item.length && !$(item[0].parentElement).attr("id")) { return item; }
```

Comment:
So, I guess `$(item[0].parentElement).attr("id")` means that if the menu element does not have a parent with an id attribute it is a main menu. Otherwise, submenu.
Feels a bit sneaky and took me some time to understand. I would suggest that you add classes like `submenu-item` to distinguish them for the sake of readability.